### ISSUE TITLE
HighlightColor opacity not applied in PolygonLayer @6.4.0

### ISSUE THREAD

**Issue by danmarshall** [uber/deck.gl#2664]
#### Description
In a mouseover of a polygon, the opacity from the original fillColor is used instead of the opacity specified in the `highlightColor` property.  This bug was introduced in 6.4.0, it works correctly in 6.3.9:

6.4.0
![image](https://user-images.githubusercontent.com/11507384/52527434-57f94980-2c7d-11e9-8671-049aa4b06983.png)

6.3.9
![image](https://user-images.githubusercontent.com/11507384/52527437-5f205780-2c7d-11e9-97e0-54da7cb132ee.png)

#### Repro Steps
[Codepen 6.4.0](https://codepen.io/danmar/pen/exMmPX)
[Codepen 6.3.9](https://codepen.io/danmar/pen/rPdaqb)

Mouseover the red triangle.

```
<html>

<body>
    <script src="https://unpkg.com/deck.gl@~6.4.0/deckgl.min.js"></script>
    <div id="div1" style="height: 500px;"></div>
    <script>
        function triangle(d, fillColor) {
            return { polygon: [[0, 0], [d, 0], [0, d]], fillColor };
        }
        var deckgl = new deck.DeckGL({
            views: [new deck.OrbitView()],
            viewState: { distance: 50, rotationX: 64, rotationOrbit: 72, fov: 50, zoom: 0.2 },
            container: 'div1',
            layers: [new deck.PolygonLayer({
                coordinateSystem: deck.COORDINATE_SYSTEM.IDENTITY,
                data: [
                    triangle(100, [0, 0, 255, 255]), //blue 100%
                    triangle(-100, [255, 0, 0, 25.5]) //red at 10% opacity
                ],
                extruded: true,
                highlightColor: [0, 255, 0, 255], //green at 100% opacity
                autoHighlight: true,
                pickable: true,
                getElevation: d => 20,
                getFillColor: d => d.fillColor
            })]
        });
    </script>
</body>

</html>
```

#### Environment (please complete the following information):
 - Framework Version: 6.4.0
 - Browser Version: Chrome 72.0
 - OS: Windows 10

**Comment by danmarshall** [comment]
Persists in 6.4.1 - [Codepen](https://codepen.io/danmar/pen/RvMNOV)

**Comment by Pessimistress** [comment]
This was introduced in [this PR](https://github.com/uber/luma.gl/pull/799), where you can follow the original discussion. The change was meant to fix a use case that is not necessarily in conflict with yours.

There are arguments for both behaviors. I agree it's no longer easy for you to achieve the desired effect in your use case, which is a regression.

**Comment by danmarshall** [comment]
Thanks for the pointer @Pessimistress . My use case is to have invisible `[0,0,0,0]` polygons as "hotspots". Is the new model only a mix? If so, my color will always be mixed at 0%.

**Comment by Pessimistress** [comment]
To sum up, we have the following use cases:

| Description | Expected alpha | 6.3 (highlight.alpha) | 6.4 (object.alpha) |
| ---- | ---- | ---- | ---- |
| Basic - `highlight.alpha: 1`, `object.alpha: 1` | `1` | `1` | `1` |
| kepler.gl - `highlight.alpha: 0.6`, `object.alpha: 1` | `1` | `0.6` | `1` |
| Extra hit area - `highlight.alpha: 1`, `object.alpha: 0` | `0` | `1` | `0` |
| @danmarshall's invisible hotspots 	- `highlight.alpha: 1`, `object.alpha: 0` | `1` | `1` | `0` |

Essentially we need two separate settings to control color and alpha blending. There really isn't a perfect solution unless we stop overloading `highlight.alpha` and introduce a new prop.

Alternatively, as a compromise, we can switch to use blended alpha (`highlightA + objectA * (1.0 - highlightA)`). It will continue to work for use case 1, 2; fix 4; and partially break 3. However it's probably still better than the current behavior, where the highlight settings do not have any control over the result alpha.

@jianhuang01 @ibgreen

**Comment by ibgreen** [comment]
> introduce a new prop.

While we certainly want to be conservative with adding new props, I think it might be justified here. 

I am not 100% clear on how @danmarshall 's hotspots work, but if this is turning out to be one of those features that can be used to solve interesting/tricky interaction problems in applications, then we might want to make sure we don't cripple it by trying to be too efficient.

Do you already have name/semantics for the new prop in mind?

**Comment by Pessimistress** [comment]
@danmarshall  This should be fixed in 6.4.2
